includes/class-wc-payments-woopay-direct-checkout.php

```diff
@@ -94,11 +127,14 @@ public function scripts() {
 	 * - The current page is the cart page.
 	 * - The current page has a cart block.
 	 * - The current page has the blocks mini cart widget, i.e 'woocommerce_blocks_cart_enqueue_data' has been fired.
+	 * - The current page has the cart fragments script enqueued. which is enqueued by the shortcode mini cart widget.
 	 *
 	 * @return bool True if the scripts should be enqueued, false otherwise.
 	 */
 	private function should_enqueue_scripts(): bool {
-		return $this->is_cart_page() || did_action( 'woocommerce_blocks_cart_enqueue_data' ) > 0;
+		return $this->is_cart_page()
+			|| did_action( 'woocommerce_blocks_cart_enqueue_data' ) > 0
+			|| ( wp_script_is( 'wc-cart-fragments', 'enqueued' ) && ! $this->is_checkout_page() );
```

Comment:
This entire condition could probably be turned into `! $this->is_checkout_page`, however, I didn't want to leave it too loose because we don't know what kind of customizations the merchants are applying to their themes. I went for a safer approach and added a condition to check whether the `wc-cart-fragments` is enqueued ([ref](https://github.com/woocommerce/woocommerce/blob/35cd4518bb33524c5c7209dc96f1adaf46a74858/plugins/woocommerce/includes/widgets/class-wc-widget-cart.php#L59)) and it's not the checkout page.

I also tried a similar approach used for the Blocks mini cart by doing `did_action( 'woocommerce_widget_shopping_cart_buttons' ) > 0` ([ref](https://github.com/woocommerce/woocommerce/blob/8ee74ede0ebc1957089af08207ae2a1a8250610e/plugins/woocommerce/templates/cart/mini-cart.php#L92)) but that always return `0` since that action is not fired before the `wp_footer` hook.